pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -774,6 +774,9 @@ monitorLoop:
 				if strings.Contains(passedErr.Error(), "canceled by client") {
 					abortStatus = v1.MigrationAbortSucceeded
 				}
+				if strings.Contains(passedErr.Error(), "client socket is closed") {
```

Comment:
@maya-r I'm not that this is a good indication of a user requested migration cancelation.
The only error that would be raised because of cancelation is "canceled by client" [1]
I suppose "client socket is closed"  error could occur because of various reasons - perhaps libvirtd didn't respond.

In which context did you see this error?

[1] https://github.com/libvirt/libvirt/blob/master/src/qemu/qemu_migration.c#L1731

Reply:
It is in this run: https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/5092/pull-kubevirt-e2e-k8s-1.18/1374724631486795776

in 1_kubevirt-test-default1_virt-launcher-testvmi-jn9sr-8rzgp-compute.log

```
{"component":"virt-launcher","level":"error","msg":"operation aborted: migration out: canceled by client","pos":"qemuMigrationSrcNBDStorageCopy:1154","subcomponent":"libvirt","thread":"30","timestamp":"2021-03-24T17:07:17.795000Z"}
...
{"component":"virt-launcher","level":"error","msg":"internal error: client socket is closed","pos":"virNetClientSendInternal:2114","subcomponent":"libvirt","thread":"30","timestamp":"2021-03-24T17:07:18.026000Z"}
```

And I guess we end up seeing the second error first, break, and never see the first one.

Reply:
(I figured it'd be best to discuss the problem over a PR rather than private messages. Perhaps I should label it a draft)

Reply:
I think the problem here is different. 

If you look at the log:

```
{"component":"virt-launcher","kind":"","level":"info","msg":"Live migration abort succeeded","name":"testvmi-jn9sr","namespace":"kubevirt-test-default1","pos":"manager.go:930","timestamp":"2021-03-24T17:07:17.302051Z","uid":"e8f1e215-2a98-462a-8802-0f313449e6b3"}
```

This ^^^ shows us that we've got the correct error message and acted correctly.

```
{"component":"virt-launcher","level":"info","msg":"kubevirt domain status: Running(1):Unknown(1)","pos":"client.go:262","timestamp":"2021-03-24T17:07:17.304118Z"}
{"component":"virt-launcher","level":"info","msg":"Domain name event: kubevirt-test-default1_testvmi-jn9sr","pos":"client.go:377","timestamp":"2021-03-24T17:07:17.306193Z"}

{"component":"virt-launcher","kind":"","level":"error","msg":"Unable to post migration results to libvirt after multiple tries","name":"testvmi-jn9sr","namespace":"kubevirt-test-
default1","pos":"manager.go:264","reason":"Migration abort is in progress","timestamp":"2021-03-24T17:07:17.308374Z","uid":"e8f1e215-2a98-462a-8802-0f313449e6b3"}
```

However, the problem is here ^^^ that we didn't manage to write this information into the domain metadata.

```
{"component":"virt-launcher","kind":"","level":"info","msg":"Live migration as been aborted","name":"testvmi-jn9sr","namespace":"kubevirt-test-default1","pos":"server.go:129","timestamp":"2021-03-24T17:07:17.308433Z","uid":"e8f1e215-2a98-462a-8802-0f313449e6b3"}
{"component":"virt-launcher","level":"error","msg":"operation aborted: migration out: canceled by client","pos":"qemuMigrationSrcNBDStorageCopy:1154","subcomponent":"libvirt","thread":"30","timestamp":"2021-03-24T17:07:17.795000Z"}
```

Reply:
That's the reason we saw this `internal error: client socket is closed` from libvirt, because it has restarted or was blocked and didn't manage to send a keepalive.

Reply:
We should investigate this further to understand what's going on with libvirt. Perhaps we should try to reproduce this with libvirt debug logs enabled.

Reply:
ah! I know what the problem is... :facepalm:

Reply:
I've posted #5323 - let's see.